src/providers/pkcs11/key_management.rs

```diff
@@ -444,9 +444,14 @@ impl Provider {
             // See ECPoint in [SEC1](https://www.secg.org/sec1-v2.pdf). PKCS11 mandates using
             // [ANSI X9.62 ECPoint](https://cryptsoft.com/pkcs11doc/v220/group__SEC__12__3__3__ECDSA__PUBLIC__KEY__OBJECTS.html),
             // however SEC1 is an equivalent spec.
-            let key_data: OctetStringAsn1 = picky_asn1_der::from_bytes(&data)
-                .map_err(|_| ResponseStatus::PsaErrorGenericError)?;
-            Ok(key_data.0)
+            let parsed_data = picky_asn1_der::from_bytes::<OctetStringAsn1>(&data);
+            match parsed_data {
+                Ok(key_data) => Ok(key_data.0),
+                // Some PKCS#11 implementations provide the EC_POINT as raw bytes and fail to wrap in ASN.1 OctetString as
+                // mandated by the spec. If the ASN.1 parse fails, then we assume that the raw bytes are the EC_POINT data,
+                // and return those instead of failing, trading off strictness for broader interoperability.
+                Err(_) => Ok(data),
```

Comment:
Could we be sure of getting `InvalidData` specifically? Might the actual error depend on exactly what the bytes look like, especially at the start of the stream? I might be able to experiment for my current error case, but I think I'd then be less confident about the patch succeeding in all cases, because the parser Is essentially being given garbage in these cases.

Another way to improve this patch might be to validate that the raw (unparsed) data is indeed a valid EC_POINT, but I'm not sure whether we can do that? Can we infer that from its length or from the content?

Reply:
> Another way to improve this patch might be to validate that the raw (unparsed) data is indeed a valid EC_POINT, but I'm not sure whether we can do that? Can we infer that from its length or from the content?

We can, see the description for public ECC keys [here](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_export_public_key.html#description) (and you can see some checks in `handle_ecc_public_import_attrib`, though we don't check that the first byte is `0x04`.

> Could we be sure of getting InvalidData specifically? Might the actual error depend on exactly what the bytes look like, especially at the start of the stream? I might be able to experiment for my current error case, but I think I'd then be less confident about the patch succeeding in all cases, because the parser Is essentially being given garbage in these cases.

That's fair, perhaps we can resort to a length check and checking that the first byte is 4 and call it a day!

Reply:
Looks like the first byte is only `0x04` for some curve types, not all - this could all get a bit messy,

Reply:
Perhaps what we need to do is refactor some of the checks in `handle_ecc_public_import_attrib` and directly re-use them, possibly adding the `0x04` check for the curve types where that is applicable. But I think that's got to be a separate piece of work, beyond this PR.

Reply:
This PR will at least fix the interop issue that we have (on Layerscape and possibly other systems) and it doesn't regress anything else.

Reply:
Cool, then we can get this one merged and either raise a new issue for the cleanup or repurpose the existing issue for it (?)

Reply:
I'd raise a new issue, I think, so that it can be prioritised separately. The evidence we have so far is that implementations that don't provide the ASN.1 wrapping do at least provide valid key data - that's true for Layerscape and also true in another similar case that I've seen. So these extra checks would be ultra-paranoid - not necessarily a bad thing, but the risk is that the checks themselves could have bugs and cause us to spuriously reject data that is, in fact, valid. That's the main thing I'm worried about, and why I don't really want to overload this PR with anything like that.